Remove dead code in ddp.{h, cpp}

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37990 Remove dead code in ddp.{h, cpp}**

The code in `ddp.{h, cpp}` and the corresponding pybind implementations are no longer used. The pybinded calls were all private APIs and only ran in unittests, so we should remove these unused APIs.

https://github.com/pytorch/pytorch/pull/20234 from a year ago also mentioned that we should delete `_dist_broadcast_coalesced`

Verified that all tests pass with cuda by running `test_c10d` on a gpu-enabled machine.

Differential Revision: [D21443879](https://our.internmc.facebook.com/intern/diff/D21443879/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21443879/)!

rohan-varma committed:
Remove dead code in ddp.{h, cpp}

The code in `ddp.{h, cpp}` and the corresponding pybind implementations are no longer used. The pybinded calls were all private APIs and only ran in unittests, so we should remove these unused APIs.

https://github.com/pytorch/pytorch/pull/20234 from a year ago also mentioned that we should delete `_dist_broadcast_coalesced`

Verified that all tests pass with cuda by running `test_c10d` on a gpu-enabled machine.

Differential Revision: [D21443879](https://our.internmc.facebook.com/intern/diff/D21443879/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21443879/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "Remove dead code in ddp.{h, cpp}"


The code in `ddp.{h, cpp}` and the corresponding pybind implementations are no longer used. The pybinded calls were all private APIs and only ran in unittests, so we should remove these unused APIs.

https://github.com/pytorch/pytorch/pull/20234 from a year ago also mentioned that we should delete `_dist_broadcast_coalesced`

Verified that all tests pass with cuda by running `test_c10d` on a gpu-enabled machine.

Differential Revision: [D21443879](https://our.internmc.facebook.com/intern/diff/D21443879/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21443879/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "Remove dead code in ddp.{h, cpp}"


The code in `ddp.{h, cpp}` and the corresponding pybind implementations are no longer used. The pybinded calls were all private APIs and only ran in unittests, so we should remove these unused APIs.

https://github.com/pytorch/pytorch/pull/20234 from a year ago also mentioned that we should delete `_dist_broadcast_coalesced`

Verified that all tests pass with cuda by running `test_c10d` on a gpu-enabled machine.

Differential Revision: [D21443879](https://our.internmc.facebook.com/intern/diff/D21443879/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21443879/)!

[ghstack-poisoned]

mrshenli reviewed (approved):
It's great to see getting rid of dead code! The failed tests doesn't seem to be broken by this PR, but can you double check if they are flaky instead of consistently failing by running those locally

I created #38015 and #38016 to track.

rohan-varma commented:
@mrshenli Actually it looks like some of the test issues are related. In `test_distributed.py`, we define an exit code of 77 for some specific skip condition, but in `common_distributed` that code is for something else. This is a duplication issue now that test_distributed uses the utils in common_distributed. Now in common_distributed I removed the 77 skip code, messing things up. I'll send a PR deduping these exit codes below this.

rohan-varma committed:
Update on "Remove dead code in ddp.{h, cpp}"


The code in `ddp.{h, cpp}` and the corresponding pybind implementations are no longer used. The pybinded calls were all private APIs and only ran in unittests, so we should remove these unused APIs.

https://github.com/pytorch/pytorch/pull/20234 from a year ago also mentioned that we should delete `_dist_broadcast_coalesced`

Verified that all tests pass with cuda by running `test_c10d` on a gpu-enabled machine.

Differential Revision: [D21443879](https://our.internmc.facebook.com/intern/diff/D21443879/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21443879/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "Remove dead code in ddp.{h, cpp}"


The code in `ddp.{h, cpp}` and the corresponding pybind implementations are no longer used. The pybinded calls were all private APIs and only ran in unittests, so we should remove these unused APIs.

https://github.com/pytorch/pytorch/pull/20234 from a year ago also mentioned that we should delete `_dist_broadcast_coalesced`

Verified that all tests pass with cuda by running `test_c10d` on a gpu-enabled machine.

Differential Revision: [D21443879](https://our.internmc.facebook.com/intern/diff/D21443879/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21443879/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "Remove dead code in ddp.{h, cpp}"


The code in `ddp.{h, cpp}` and the corresponding pybind implementations are no longer used. The pybinded calls were all private APIs and only ran in unittests, so we should remove these unused APIs.

https://github.com/pytorch/pytorch/pull/20234 from a year ago also mentioned that we should delete `_dist_broadcast_coalesced`

Verified that all tests pass with cuda by running `test_c10d` on a gpu-enabled machine.

Differential Revision: [D21443879](https://our.internmc.facebook.com/intern/diff/D21443879/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21443879/)!

[ghstack-poisoned]
